src/__tests__/index.js

```diff
@@ -251,4 +252,45 @@ import LeftPad from 'left-pad-js'
   assert.match(container.innerHTML, 'this is left pad')
 })
 
+test('require from current directory', async () => {
+  const mdxSource = `
+# Title
+
+import {Sample} from './other/sample-component'
+
+<Sample />
+
+![A Sample Image](./other/150.png)
+`.trim()
+
+  const {code} = await bundleMDX(mdxSource, {
+    cwd: process.cwd(),
+    xdmOptions: (vFile, options) => {
+      options.remarkPlugins = [
+        remarkMdxImages
+      ]
+
+      return options
+    },
+    esbuildOptions: options => {
+      options.loader = {
+        ...options.loader,
+        '.png': 'dataurl',
+      }
+
+      return options
+    },
+  })
+
+  const Component = getMDXComponent(code)
+
+  const {container} = render(React.createElement(Component))
+
+  assert.match(container.innerHTML, 'Sample!')
+  // Test that the React components image is imported correctly.
+  assert.match(container.innerHTML, 'img src="data:image/png')
+  // Test that the markdowns image is imported correctly.
+  assert.match(container.innerHTML, 'img alt="A Sample Image" src="data:image/png')
```

Comment:
This test is using [remark-mdx-images](https://www.npmjs.com/package/remark-mdx-images) to change images to imports so esbuild bundles them. It's a proof-of-concept for co-locating images inside your posts directory.

I'll document this approach once I have something working on my site but I'll need to use this PR's version of mdx-bundler to do that.